internal/database/repository/partitions.go

```diff
@@ -88,3 +91,19 @@ func (s *PostgresRepository) GetAllPartitions(ctx context.Context) ([]*dao.Parti
 	}
 	return partitions, nil
 }
+
+func (s *PostgresRepository) DeletePartitions(ctx context.Context, partitions []*model.PartitionInfo) error {
+	partitionNames := make([]string, len(partitions))
+	for i, p := range partitions {
+		partitionNames[i] = p.Name
+	}
+
+	deletedAt := time.Now().Unix()
+	query := `UPDATE partitions SET deleted_at = $1 WHERE name = ANY($2)`
```

Comment:
You should probably target these partitions by ID. Imagine that we had a partition named "example". Then we removed it. Then we re-installed it. Then we remove it again. Currently, in the database, we have multiple partition records with the name "example", but only one of them should have `deleted_at` set to now.